test/soca/CMakeLists.txt

```diff
@@ -32,3 +36,95 @@ set_tests_properties(
         test_gdasapp_soca_nsst_increment_to_mom6
 PROPERTIES
 ENVIRONMENT "data=${TESTDATA};PYTHONPATH=${PROJECT_BINARY_DIR}/ush:$ENV{PYTHONPATH}")
+
+
+###########################################################################
+# bufr to ioda tests:
+###########################################################################
+
+# prepare a test.yaml file from test.yaml.in by replacing
+# placeholder patterns __BUFRINPUTDIR__ and __IODAOUTPUTDIR__
+# with actual directory paths
+function(CREATE_CONFIG_FILE test_config_in test_config_out bufr_input_dir ioda_output_dir)
+	file(READ "${test_config_in}" file_content)
+	string(REPLACE "__BUFRINPUTDIR__" "\"${bufr_input_dir}\"" temp_content "${file_content}")
+	string(REPLACE "__IODAOUTPUTDIR__" "\"${ioda_output_dir}\"" temp_content2 "${temp_content}")
+	file(WRITE "${test_config_out}" "${temp_content2}")
+endfunction()
+
+
+# set(TEST_WORKING_DIR "/scratch1/NCEPDEV/stmp2/Edward.Givelberg/RUNDIRS/GFSv17-3DVAR-C384mx025/prepoceanobs.114138")
+set(TEST_WORKING_DIR ${PROJECT_BINARY_DIR}/test/soca)
+message("bufr to ioda: TEST_WORKING_DIR = ${TEST_WORKING_DIR}")
+
+set(MARINE_BUFR2IODA_DIR ${PROJECT_SOURCE_DIR}/ush/ioda/bufr2ioda/marine)
+set(MARINE_BUFR2IODA_DIR ${MARINE_BUFR2IODA_DIR}/b2i)
+set(CONFIG_DIR ${PROJECT_SOURCE_DIR}/test/soca/testinput)
+set(TESTREF_DIR ${PROJECT_SOURCE_DIR}/test/soca/testref)
+
+# find the staging directory for bufr test input files ${BUFR_TEST_DIR}
+# bufr input files on orion:
+set(BUFR_TEST_DIR
+	"/work/noaa/da/marineda/gfs-marine/data/obs/ci/bufr"
+	# "/work/noaa/da/marineda/gfs-marine/data/obs/ci/obs"
+	# ${TEST_WORKING_DIR}			# temporary fix
+)
+if (NOT EXISTS ${BUFR_TEST_DIR})
+	# bufr input files on hera:
+	set(BUFR_TEST_DIR
+		"/scratch1/NCEPDEV/da/common/ci/bufr"
+		# "/scratch1/NCEPDEV/da/common/ci/obs"
+		# ${TEST_WORKING_DIR}			# temporary fix
+	)
+endif()
+if (NOT EXISTS ${BUFR_TEST_DIR})
+	message(FATAL_ERROR "staging directory for BUFR test files ${BUFR_TEST_DIR} not found")
```

Comment:
Since apparently you only implemented the test for Hera, replace with a warning for now.